source/pervasive/atomics.rs

```diff
@@ -1,89 +1,464 @@
-use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::atomic::{
```

Comment:
Sorry I only spotted this now, but can you rename `pervasive::atomics` to `pervasive::atomic` (singular), to match the rust std naming convention?

(We should also probably rename `invariants` to `invariant` and maybe move both plus `cell` to a new `sync` module, but not in this PR.)